src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java

```diff
@@ -274,7 +274,7 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnul
 
         for (Condition condition : conditions) {
             Result validationResult = condition.validate(buildActionDTO, compareDTO, launcher, listener);
-            Pair<String, String> pair = Pair.with(condition.getDescriptor().getDisplayName(), validationResult.toString());
+            Pair<String, String> pair = Pair.with(condition.describeResult(), validationResult.toString());
             conditionNamesAndResults.add(pair);
             run.setResult(validationResult);
             listener.getLogger().println(String.format("[CodeSonar] '%s' marked the build as %s", condition.getDescriptor().getDisplayName(), validationResult.toString()));
```

Comment:
By the intentions expressed in the PR description, I think you would want to modify the `lister.getLogger().println()` call instead of the `Pair` construction.  The data for the `Pair` gets saved away someplace, but it does not appear to be logged.  However, if your goal is to log the condition output, then the log printer a few lines below seems like the right place.

Reply:
The intention was to replace the text on the status page of a Jenkins build using the plugin with something more informative.
Currently, the plugin only presents the ```NAME``` of the condition class and the validation result. This does not help the user see why the condition had a certain result, nor does it help distinguish multiple instances of the same condition class from one another.
For example, if the job has two instances of the WarningCountAbsoluteSpecifiedScoreAndHigherCondition, one for warnings of score >= 21 and one for warnings of score >= 56, then it is not clear on the Jenkins build page which of these two have failed, or by how much.

Reply:
I see, thank you for the clarification